supervisor/const.py

```diff
@@ -382,12 +382,27 @@
 ROLE_ALL = [ROLE_DEFAULT, ROLE_HOMEASSISTANT, ROLE_BACKUP, ROLE_MANAGER, ROLE_ADMIN]
 
 
+class AddonBootConfig(StrEnum):
+    """Boot mode config for the add-on."""
+
+    AUTO = "auto"
+    MANUAL = "manual"
+    MANUAL_FORCED = "manual_forced"
+
+
 class AddonBoot(StrEnum):
     """Boot mode for the add-on."""
 
     AUTO = "auto"
     MANUAL = "manual"
 
+    @classmethod
+    def _missing_(cls, value: str) -> Self | None:
+        """Convert 'forced' config values to their counterpart."""
+        if value == AddonBootConfig.MANUAL_FORCED:
+            return AddonBoot.MANUAL
```

Comment:
Why is this needed exactly? The config will use `AddonBootConfig` which know the new value, and the API we'll not change no? :thinking:

Reply:
Because the `boot` option is currently set to fallback on the config value if the user has not overwritten it by changing the setting. And the config value now can have a new option which does not make sense for `boot`. Because an addon is either booted automatically at startup or manually on demand, this third option is not really a new boot option it simply prevents user changes to the boot option.

So my two options were to add an if statement here to convert `manual_forced` into `manual`:
https://github.com/home-assistant/supervisor/blob/24050f594cc09ac8cc23fb05910da151166aa732/supervisor/addons/model.py#L159-L161

Or add the if statement there in the enum itself. The latter option seemed cleaner to me, especially if we do decide to add an `auto_forced` in the future.